### DISCUSSION TITLE
Cannot find libfaketime.so.1 using compiler.find_library()

### DISCUSSION THREAD

**Discussion by andy5995** [mesonbuild/meson#13806] [Q&A] [answered]
Where I'm testing locally, libfaketime.so.1 is in /usr/lib/faketime. This code is not finding it:

```
faketime_lib = cc.find_library(
  'faketime',
  dirs: [
    '/usr/lib/faketime',
    '/lib/x86_64-linux-gnu/faketime',
    '/lib/aarch64-linux-gnu/faketime'
    ],
  required: true
  )
```

After a little more testing, I see now that meson will find it if I create a libfaketime.so symlink to the so.1 file.

I wonder why on Manjaro and Ubuntu Linux (the only two I've tried this on), the .so symlink isn't created when  installing the package.

I am trying to use faketime for a test setup:

```
add_test_setup(
    'epochalypse',
    env: [
        #'LD_PRELOAD=' + faketime_lib.full_path(),
        'LD_PRELOAD=/usr/lib/faketime/libfaketime.so.1',
        'FAKETIME=+14y'
    ]
)
```

Any suggestions as to a different approach? Maybe it would be best if I bundled the [faketime source](https://github.com/wolfcw/libfaketime) with my project.

**Comment by jpakkane** [comment]
You need to install the corresponding `-dev` package to get the symlink. This applies to all shared library dependencies.

**Reply by andy5995** [reply]
There is no corresponding '-dev' package available for Ubuntu or Manjaro.

**Comment by jpakkane** [comment]
FWICT you are not supposed to link against faketime, instead you have to set `LD_PRELOAD` when launching your program. `find_library` only finds libraries you can directly link against.

**Reply by andy5995** [reply]
@jpakkane You are correct. I set LD_PRELOAD in the call to add_test_setup() (example in original post). I was hoping I could get the path somehow when the library was found.

Since I'm not actually linking against libfaketime, in this case, it might be best if I were using a function that just finds *any* file.

**Comment by eli-schwartz** [comment, marked as answer]
Does the `faketime` program wrapper not work for your use case?

**Reply by andy5995** [reply]
I didn't think of it. Yes it does. Here's my working code:

```meson
faketime = find_program('faketime', required : false)
if faketime.found()
  add_test_setup(
    'epochalypse',
    exe_wrapper : [faketime, '-f', '+14y']
  )
endif
```

Thanks, again!
